cmd/addvalidators.go

```diff
@@ -65,15 +74,23 @@ func newAddValidatorsCmd(runFunc func(context.Context, addValidatorsConfig) erro
 // bindAddValidatorsFlags binds command line flags for the `add-validators` command.
 func bindAddValidatorsFlags(cmd *cobra.Command, config *addValidatorsConfig) {
 	cmd.Flags().IntVar(&config.NumVals, "num-validators", 1, "The count of new distributed validators to add in the cluster.")
-	cmd.Flags().StringVar(&config.Lockfile, "lock-file", ".charon/cluster-lock.json", "The path to the legacy cluster lock file defining distributed validator cluster.")
-	cmd.Flags().StringSliceVar(&config.EnrPrivKeyfiles, "private-key-files", nil, "Comma separated list of paths to charon enr private key files. This should be in the same order as the operators, ie, first private key file should correspond to the first operator and so on.")
 	cmd.Flags().StringSliceVar(&config.FeeRecipientAddrs, "fee-recipient-addresses", nil, "Comma separated list of Ethereum addresses of the fee recipient for each new validator. Either provide a single fee recipient address or fee recipient addresses for each validator.")
 	cmd.Flags().StringSliceVar(&config.WithdrawalAddrs, "withdrawal-addresses", nil, "Comma separated list of Ethereum addresses to receive the returned stake and accrued rewards for each new validator. Either provide a single withdrawal address or withdrawal addresses for each validator.")
+	cmd.Flags().StringVar(&config.Lockfile, "lock-file", ".charon/cluster-lock.json", "The path to the legacy cluster lock file defining distributed validator cluster.")
+	cmd.Flags().StringVar(&config.ClusterManifestFile, "manifest-file", ".charon/cluster-manifest.pb", "The path to the cluster manifest file.")
+	cmd.Flags().StringVar(&config.ClusterBackupDir, "cluster-backup-dir", ".charon/cluster-backups", "The directory containing cluster manifest backup files.")
+	cmd.Flags().StringSliceVar(&config.EnrPrivKeyfiles, "private-key-files", nil, "Comma separated list of paths to charon enr private key files. This should be in the same order as the operators, ie, first private key file should correspond to the first operator and so on.")
 }
 
 func runAddValidatorsSolo(_ context.Context, conf addValidatorsConfig) (err error) {
-	// Read lock file to load mutable cluster manifest.
-	cluster, err := loadClusterState(conf)
+	// Read lock file to load mutable cluster manifest
+	cluster, err := loadClusterManifest(conf)
+	if err != nil {
+		return err
+	}
+
+	// Write backup of the current cluster manifest to disk
+	err = writeClusterBackup(conf.ClusterBackupDir, cluster)
```

Comment:
can we move this to end of this function? like write backup file when validators are added and we have a new cluster manifest. 
This is way we can avoid the situation where we have written the backup file first but the command failed after that. In that case, our existing cluster manifest will be same as the backup one and it can be confusing to the operators.

Reply:
Yeah, good point. This also ensures backups are created only when it's time to override the `cluster-manifest.pb` since we can encounter errors before getting to the final point of storing to disk.